src/ci/docker/dist-fuchsia/build-toolchain.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/bash
+# Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+# file at the top-level directory of this distribution and at
+# http://rust-lang.org/COPYRIGHT.
+#
+# Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+# http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+# <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+# option. This file may not be copied, modified, or distributed
+# except according to those terms.
+
+set -ex
+
+# Download sources
+host=https://fuchsia.googlesource.com
+git clone --depth 1 $host/magenta
```

Comment:
We tend to prefer to pin to specific commits where possible to avoid an upstream commit accidentally breaking our build, could these clones all be followed with a `cd $dir && git reset --hard $rev`? 

It should be fine to do a full clone, the Docker container is cached so we won't pay that build time on each build.